Add fallback for copy_file_range EXDEV error

Xist12gh opened the pull request:
Fixes #3654 

Add fallback to other copy routines in case of error in previous attempt.
I tried to keep it generic, but for now it targets `EXDEV` error in `copy_file_range` explicitly.

For efficiency sake, I kept the tiered approach to available kernel routines which was previously selected by compile flags.
To do the fallback in runtime I had to duplicate some code. The effort to refactor this went a bit above the scope of a bugfix. At least for now it keeps the change readily accessible for review.
Added some comments for the one trying to clean this up.

**Functional changes:**
Previously:
Try one of `copy_file_range` or `sendfile64` in this order selected during compile time, if it fails exit `tr_sys_path_copy`
Use user-space copy if neither was available during compile time

Now:
First try `copy_file_range` if available during compile time.
Try `sendfile64` if available in case of previous `EXDEV` or when `copy_file_range` wasn't available.
Then try user-space copy if neither kernel routine was available during compile time or previous `EXDEV` errno was set

ckerr reviewed (commented):
FTBFS:

/usr/home/user/buildAgent/work/1f3bff43dd55cfe9/libtransmission/file-posix.cc:491:5: error: unknown type name 'error_t'; did you mean 'errno_t'?

kisol-pw commented:
hm, as I feared, too environment specific.
well, let's not get overambitious and go with the traditional int then.
I'll push an updated when I get off work in a few hours.

Xist12gh commented:
Updated PR

* rebased to https://github.com/transmission/transmission/commit/72a16969006da9f1f7784e51ec97029c177aaa06
* switched to traditional int for errno
* fixed unneeded logging of `EXDEV` even though the fallback catches it, somehow slipped the original testing

Let's see if this works now.

If nobody beats me to it, I'll probably refactor the duplicated parts in a later PR once this one proves to run stable.
